instrumentation/runtime/doc.go

```diff
@@ -12,21 +12,80 @@
-//   runtime.go.mem.live_objects  -          Number of live objects is the number of cumulative Mallocs - Frees
-//   runtime.uptime               (ms)       Milliseconds since application was initialized
+//   process.runtime.go.cgo.calls         -          Number of cgo calls made by the current process
+//   process.runtime.go.gc.count          -          Number of completed garbage collection cycles
+//   process.runtime.go.gc.pause_ns       (ns)       Amount of nanoseconds in GC stop-the-world pauses
+//   process.runtime.go.gc.pause_total_ns (ns)       Cumulative nanoseconds in GC stop-the-world pauses since the program started
+//   process.runtime.go.goroutines        -          Number of goroutines that currently exist
+//   process.runtime.go.lookups           -          Number of pointer lookups performed by the runtime
+//   process.runtime.go.mem.heap_alloc    (bytes)    Bytes of allocated heap objects
+//   process.runtime.go.mem.heap_idle     (bytes)    Bytes in idle (unused) spans
+//   process.runtime.go.mem.heap_inuse    (bytes)    Bytes in in-use spans
+//   process.runtime.go.mem.heap_objects  -          Number of allocated heap objects
+//   process.runtime.go.mem.heap_released (bytes)    Bytes of idle spans whose physical memory has been returned to the OS
+//   process.runtime.go.mem.heap_sys      (bytes)    Bytes of heap memory obtained from the OS
+//   process.runtime.go.mem.live_objects  -          Number of live objects is the number of cumulative Mallocs - Frees
+//   runtime.uptime                       (ms)       Milliseconds since application was initialized
+//
+// The Go-1.16 release featured a new runtime/metrics package that gives formal
+// metric names to the various quantities.  This package supports the new metrics
+// under their Go-specified names by setting `WithUseGoRuntimeMetrics(true)`.
+// These metrics are documented at https://pkg.go.dev/runtime/metrics#hdr-Supported_metrics.
+//
+// The `runtime/metrics` implementation will replace the older
+// implementation as the default no sooner than January 2023. The
+// older implementation will be removed no sooner than January 2024.
+//
+// The following metrics are generated in go-1.19.
+//
+// Name                                                    Unit         Instrument
+// -----------------------------------------------------------------------------
+// process.runtime.go.cgo.go-to-c-calls                    {calls}      Counter[int64]
+// process.runtime.go.gc.cycles.automatic                  {gc-cycles}  Counter[int64]
+// process.runtime.go.gc.cycles.forced                     {gc-cycles}  Counter[int64]
+// process.runtime.go.gc.cycles.total                      {gc-cycles}  Counter[int64]
+// process.runtime.go.gc.heap.allocs.bytes                 (*)          Counter[int64]
+// process.runtime.go.gc.heap.allocs.objects               (*)          Counter[int64]
```

Comment:
Here is a note about prometheus naming I forgot in our earlier conversation, which may help us: Counts of a thing (e.g. objects, or requests) are [considered unit-less in prometheus](https://prometheus.io/docs/practices/naming/#metric-names), and thus aren't impacted by unit naming conventions.  For example, in this case, we could have the two metrics:

* `process.runtime.go.gc.heap.allocs` (unit = bytes), and 
* `process.runtime.go.gc.heap.allocs.objects` (unit = {objects})

which would produce:

* `process_runtime_go_gc_heap_allocs_bytes_total` (unit = bytes)
* `process_runtime_go_gc_heap_allocs_objects_total` (no unit)

The `{objects}` unit isn't round-tripped, but that isn't possible given prom's current unit standards, and also is better (I think) than no units at all.

Reply:
See https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/pkg/translator/prometheus#metric-name for the current (alpha) strategy for unit handling in the prometheus receiver/exporter.

Reply:
Nice, will do.

Reply:
Done.